website/docs/src/content/cloud/observability.mdx

```diff
@@ -0,0 +1,269 @@
+  const parsedData = JSON.parse(data);
+  const enrichedData = await fetchAdditionalInfo(parsedData);
+  return enrichedData;
+});
+
+// Create a component to trace response generation
+const GenerateResponse = gensx.Component(
+  "GenerateResponse",
+  async ({ data }) => {
+    // This will appear as a separate node in the trace
+    return `Processed result: ${JSON.stringify(data)}`;
+  },
+);
+```
+
+## Secrets scrubbing
+
+GenSX enables you to configure which input props and outputs are marked as secrets and redacted from traces. Scrubbing happens locally before traces are sent to GenSX Cloud.
+
+### How secrets scrubbing works
+
+When a component executes, GenSX automatically:
+
+1. Identifies secrets in component props and outputs
+2. Replaces these secrets with `[secret]` in the trace data
+3. Propagates secret detection across the entire component hierarchy
+
+Even if a secret is passed down through multiple components, it remains scrubbed in all traces.
+
+### Marking secrets in component props
+
+To mark specific props as containing secrets:
+
+```tsx
+import * as gensx from "@gensx/core";
+
+const AuthenticatedClient = gensx.Component(
+  "AuthenticatedClient",
+  ({ apiKey, endpoint, query, credentials }) => {
+    // Use apiKey securely, knowing it won't appear in traces
+    return fetchData(endpoint, query, apiKey, credentials);
+  },
+  {
+    // Mark these props as containing sensitive data
+    secretProps: ["apiKey", "credentials.privateKey"],
+  },
+);
+```
+
+The `secretProps` option can specify both top-level props and nested paths using dot notation.
+
+### Marking component outputs as secrets
+
+For components that might return sensitive information, you can mark the entire output as sensitive:
+
+```tsx
+const GenerateCredentials = gensx.Component(
+  "GenerateCredentials",
+  async ({ userId }) => {
+    // This entire output will be marked as secret
+    return {
+      accessToken: "sk-1234567890abcdef",
+      refreshToken: "rt-0987654321fedcba",
+      expiresAt: Date.now() + 3600000,
+    };
+  },
+  {
+    secretOutputs: true,
+  },
+);
+```
+
+When `secretOutputs` is set to `true`, the entire output object or value will be treated as sensitive and masked in traces.
+
+### Secrets propagation
+
+Secrets are automatically propagated through the component hierarchy:
+
+```tsx
+const ParentComponent = gensx.Component(
+  "ParentComponent",
+  async ({ apiKey, query }) => {
+    // apiKey is marked as secret
+    const results = await ChildComponent.run({
+      query,
+      headers: { authorization: `Bearer ${apiKey}` },
+    });
+
+    // apiKey is also scrubbed here in the trace,
+    // even though ChildComponent doesn't explicitly mark it
+    return processResults(results, apiKey);
+  },
+  {
+    secretProps: ["apiKey"],
+  },
+);
+```
+
+Secrets remain scrubbed throughout the execution chain, even if they're transformed or embedded in different objects.
```

Comment:
is this section necessary? I'm not sure if it adds any clarity beyond this sentence from above:  "Even if a secret is passed down through multiple components, it remains scrubbed in all traces."

I personally found this sentence very confusing: Secrets are automatically propagated through the component hierarchy:

Reply:
removed.